File: calculator.php
<?php
// PHP - 阶乘计算
function factorial($n) {
    if ($n <= 1) return 1;
    return $n * factorial($n - 1);
}

try {
    $lines = file('input.txt', FILE_IGNORE_NEW_LINES | FILE_SKIP_EMPTY_LINES);
    $output = "PHP阶乘计算:\n";
    
    foreach ($lines as $index => $line) {
        $numbers = array_map('intval', explode(' ', trim($line)));
        if (count($numbers) >= 2) {
            $a = abs($numbers[0]) % 10; // 限制大小避免溢出
            $b = abs($numbers[1]) % 10;
            $factA = factorial($a);
            $factB = factorial($b);
            $output .= sprintf("行%d: %d! = %d, %d! = %d\n", 
                $index + 1, $a, $factA, $b, $factB);
        }
    }
    
    file_put_contents('output.txt', $output . "\n", FILE_APPEND);
    echo "PHP阶乘计算完成!\n";
} catch (Exception $e) {
    echo "PHP计算错误: " . $e->getMessage() . "\n";
}
?>

File: calculator.pl
#!/usr/bin/perl
# Perl - 最大值查找
use strict;
use warnings;

sub max {
    my ($a, $b) = @_;
    return $a > $b ? $a : $b;
}

open(my $in, '<', 'input.txt') or die "无法打开输入文件: $!";
my @lines = <$in>;
close($in);

my $output = "Perl最大值查找:\n";
my $line_num = 1;

foreach my $line (@lines) {
    chomp $line;
    my @numbers = split(/\s+/, $line);
    if (@numbers >= 2) {
        my $max_val = max($numbers[0], $numbers[1]);
        $output .= "行$line_num: max($numbers[0], $numbers[1]) = $max_val\n";
    }
    $line_num++;
}

open(my $out, '>>', 'output.txt') or die "无法打开输出文件: $!";
print $out $output . "\n";
close($out);

print "Perl最大值查找完成!\n";